lib/template/tests/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-add_unit_test(LIBTEST TARGET test_template_compile)
+add_unit_test(LIBTEST CRITERION test_template_compile)
```

Comment:
The compilation error is because you removed the `TARGET` from here, also does this test require `LIBTEST` ?

Reply:
I think it is required because it fails in cmake otherwise.

Reply:
Thanks. You are right because of that it is needed.